File: dSoak-Common-CSharp/MessageTesting/BuyBalloonTester.cs
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using Messages;
using SharedObjects;

namespace MessageTesting
{
    [TestClass]
    public class BuyBalloonTester
    {
        [TestInitialize]
        public void Setup()
        {
            MessageNumber.LocalProcessId = 100;
        }

        [TestMethod]
        public void BuyBalloon_CheckEverything()
        {
            BuyBalloon msg1 = new BuyBalloon();
            Assert.IsNotNull(msg1.MessageNr);
            Assert.AreEqual(100, msg1.MessageNr.ProcessId);
            Assert.IsTrue(msg1.MessageNr.SeqNumber > 0);
            Assert.AreEqual(msg1.MessageNr, msg1.ConvId);

            List<Penny> pennies = new List<Penny> { new Penny(), new Penny(), new Penny() };
            BuyBalloon msg2 = new BuyBalloon() { Pennies = pennies };
            Assert.IsNotNull(msg2.MessageNr);
            Assert.AreEqual(100, msg2.MessageNr.ProcessId);
            Assert.AreEqual(msg1.MessageNr.SeqNumber + 1, msg2.MessageNr.SeqNumber);
            Assert.AreEqual(msg2.MessageNr, msg2.ConvId);
            Assert.IsNotNull(msg2.Pennies);
            Assert.AreSame(pennies, msg2.Pennies);

            byte[] bytes = msg2.Encode();
            string tmp = Encoding.ASCII.GetString(bytes);

            Message msg3 = Message.Decode(bytes);
            Assert.IsTrue(msg3 is BuyBalloon);
            BuyBalloon msg4 = msg3 as BuyBalloon;
            Assert.AreEqual(msg2.MessageNr, msg4.MessageNr);
            Assert.AreEqual(msg2.ConvId, msg4.ConvId);
            Assert.AreEqual(msg2.Pennies.Count, msg4.Pennies.Count);
        }
    }
}

File: dSoak-Common-Java/src/MessageTester/BuyBalloonTester.java
package MessageTester;

import static org.junit.Assert.*;

import java.io.IOException;
import java.security.NoSuchAlgorithmException;
import java.util.ArrayList;

import org.junit.Test;

import Messages.BuyBalloon;
import Messages.Message;
import SharedObject.MessageNumber;
import SharedObject.Penny;

public class BuyBalloonTester {

	@Test
	public void test_EveryThing() throws NoSuchAlgorithmException, IOException, ClassNotFoundException
	{
		MessageNumber.LocalProcessId = 100;
		
		BuyBalloon msg1 = new BuyBalloon();
		assertNotNull(msg1.MessageNr);
		assertEquals(100, msg1.MessageNr.ProcessId);
		assertTrue(msg1.MessageNr.SeqNumber > 0);
		assertEquals(msg1.MessageNr, msg1.ConvId);
		
		ArrayList<Penny> pennies = new  ArrayList<Penny>();
		pennies.add(new Penny());
		pennies.add(new Penny());
		pennies.add(new Penny());
		
		BuyBalloon msg2 = new BuyBalloon();
		msg2.Pennies = pennies;
		
		assertNotNull(msg2.MessageNr);
		assertEquals(100, msg2.MessageNr.ProcessId);
		assertEquals(msg1.MessageNr.SeqNumber + 1 , msg2.MessageNr.SeqNumber);
		assertEquals(msg2.MessageNr, msg2.ConvId);
		assertNotNull(msg2.Pennies);
		assertSame(pennies, msg2.Pennies);
		
		byte[] bytes = msg2.Encode();
		String str = new String(bytes);
		
		Message msg3 = Message.Decode(bytes);
		assertTrue(msg3 instanceof BuyBalloon);
		BuyBalloon msg4 = (BuyBalloon) msg3;
		
		assertEquals(msg3.MessageNr, msg4.MessageNr);
		assertEquals(msg3.ConvId, msg3.ConvId);
		assertEquals(msg2.Pennies.size(), msg4.Pennies.size());
	}
}